src/design-system/components/Box/Box.tsx

```diff
@@ -0,0 +1,179 @@
+import { flatten, mapValues } from 'lodash';
+import React, { forwardRef, ReactNode, useMemo } from 'react';
+import { View } from 'react-native';
+import { negativeSpace, NegativeSpace, space, Space } from '../../layout/space';
+import {
+  BackgroundProvider,
+  BackgroundProviderProps,
+} from '../BackgroundProvider/BackgroundProvider';
+import type * as Polymorphic from './polymorphic';
+
+const fraction = (numerator: number, denominator: number) =>
+  `${(numerator * 100) / denominator}%`;
+
+const widths = {
+  '1/2': fraction(1, 2),
+  '1/3': fraction(1, 3),
+  '1/4': fraction(1, 4),
+  '1/5': fraction(1, 5),
+  '2/3': fraction(2, 3),
+  '2/5': fraction(2, 5),
+  '3/4': fraction(3, 4),
+  '3/5': fraction(3, 5),
+  '4/5': fraction(4, 5),
+  full: '100%', // eslint-disable-line prettier/prettier
+} as const;
+
+export interface BoxProps {
+  alignItems?: 'flex-start' | 'flex-end' | 'center' | 'stretch';
+  background?: BackgroundProviderProps['color'];
+  children?: ReactNode;
+  flexBasis?: 0;
+  flexDirection?: 'row' | 'row-reverse' | 'column';
+  flexGrow?: 0 | 1;
+  flexShrink?: 0 | 1;
+  flexWrap?: 'wrap';
+  justifyContent?: 'flex-start' | 'flex-end' | 'center' | 'space-between';
+  margin?: NegativeSpace;
+  marginBottom?: NegativeSpace;
+  marginHorizontal?: NegativeSpace;
+  marginLeft?: NegativeSpace;
+  marginRight?: NegativeSpace;
+  marginTop?: NegativeSpace;
+  marginVertical?: NegativeSpace;
+  padding?: Space;
+  paddingBottom?: Space;
+  paddingHorizontal?: Space;
+  paddingLeft?: Space;
+  paddingRight?: Space;
+  paddingTop?: Space;
+  paddingVertical?: Space;
+  width?: keyof typeof widths;
+}
+
+type PolymorphicBox = Polymorphic.ForwardRefComponent<typeof View, BoxProps>;
+
+/**
+ * @description Renders a single `View` element with standard styling. Any
+ * background color set via the `background` prop will be used to automatically
+ * set the color mode for nested elements. To render another element instead,
+ * you can provide the `as` prop, e.g. `<Box as={Animated.View}>`
```

Comment:
how do we care about the fact that `Animated.View` is adding two methods to `View` from TS perspective?

Reply:
The props are generic based on the `as` prop, so `Box` will pick up the prop types from `Animated.View`.

Reply:
woooow

Reply:
One downside to keep an eye on is that it doesn't seem to infer the type of the event object in event handlers, so you need to explicitly type the argument yourself 😕

Reply:
This is using an adapted version of react-polymorphic from Radix: https://github.com/radix-ui/primitives/tree/main/packages/react/polymorphic